evennia/commands/default/building.py

```diff
@@ -923,21 +923,27 @@ def func(self):
         location = caller.location
 
         # Create the new room
-        typeclass = room["option"]
-        if not typeclass:
-            typeclass = settings.BASE_ROOM_TYPECLASS
+        room_typeclass, errors = caller.get_object_typeclass(obj_type="room", typeclass=room["option"], method="dig")
+        if errors:
+            self.msg("|rError creating room:|n %s" % errors)
+        if not room_typeclass:
+            return
 
         # create room
-        new_room = create.create_object(
-            typeclass, room["name"], aliases=room["aliases"], report_to=caller
+        new_room, errors = room_typeclass.create(
+            room["name"], aliases=room["aliases"], report_to=caller, creator=caller, method="dig"
```

Comment:
Also worth, to call this argument `caller` too, for consistency. Suggest to use "cmd_dig` here to be exclicit.